dask/compatibility.py

```diff
@@ -2,15 +2,4 @@
 
 from packaging.version import parse as parse_version
 
-try:
-    from math import prod
-except ImportError:
-    # Python < 3.8
-    def prod(iterable, *, start=1):  # type: ignore
-        acc = start
-        for el in iterable:
-            acc *= el
-        return acc
-
-
 _PY_VERSION = parse_version(".".join(map(str, sys.version_info[:3])))
```

Comment:
This doesn't appear to be used anywhere (including Dask generally). Should we just drop this? Maybe this whole module?

Reply:
Good point. In general, I tend to agree that if something isn't used we should just drop it. That said, `_PY_VERSION` has been used in the past and, while it's not needed at the moment, I anticipate it will be useful again in the future (e.g. `xfail`ing a test on a specific Python version). So in this particular case I'm inclined to keep it around, but I don't think it's a huge deal either way (let me know if you'd like it removed)

Reply:
Sounds reasonable